Fix pprof recipes

milljm opened the pull request:
Use tried and true method to clean up go installs

Closes #21446

Commit:
Fix pprof recipes

Use tried and true method to clean up go installs

Closes #21446

cticenhour commented on conda/pprof/build.sh:
```diff
@@ -2,10 +2,14 @@
-go install github.com/google/pprof@c488b8fa1db3fa467bf30beb5a1d6f4f10bb1b87
+go install github.com/google/pprof@latest
+
+ls -latrh ${GOPATH}/pprof/bin
+ls -latrh ${GOPATH}/pprof/pkg
```
```suggestion
ls -latrh ${GOPATH}/bin
ls -latrh ${GOPATH}/pkg
```

Looking at the linux output, I think you need to remove the pprof dir in these lines, as it should be under `GOPATH`

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1092947/) on 3d8c2be wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/21447/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1092957/) on 3d8c2be wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>12d7d3</th>
      <th colspan=4>#21447 3d8c2b</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>83.48%</td>
      <td>83.48%</td>
      <td>+0.01%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>77440</td>
      <td>77445</td>
      <td>+5</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15326</td>
      <td>15321</td>
      <td>-5</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/21447/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/21447/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/21447/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/21447/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/21447/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/21447/coverage/contact)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/21447/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/21447/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/21447/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/21447/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/21447/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/21447/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/21447/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/21447/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/21447/coverage/navier_stokes)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/21447/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/21447/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/21447/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/21447/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/21447/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/21447/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/21447/coverage/richards)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/21447/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/21447/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/21447/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/21447/coverage/xfem)

</details>



This comment will be updated on new commits.

moosebuild commented:
Job [Apple Silicon Stack](https://civet.inl.gov/job/1092930/) on 3d8c2be : invalidated by @milljm

Commit:
Try to pin runtime deps

Coincide with moose-libmesh packages to prevent a downgrade when
this package is asked to be installed.
